src/main/java/uk/nhs/hee/trainee/details/service/PlacementService.java

```diff
@@ -112,29 +130,46 @@ public boolean deletePlacementForTrainee(String traineeTisId, String placementTi
    * @return True, or False if the placement is not in the 2024 pilot.
    */
   public boolean isPilot2024(String traineeTisId, String placementId) {
-    TraineeProfile traineeProfile = repository.findByTraineeTisId(traineeTisId);
+    Optional<Placement> optionalPlacement = pilotPreflightChecks(traineeTisId, placementId);
 
-    if (traineeProfile == null) {
-      log.info("2024 pilot: [false] trainee profile {} not found", traineeTisId);
+    if (optionalPlacement.isEmpty()) {
       return false;
     }
+    Placement placement = optionalPlacement.get();
+    LocalDate dayAfterPlacementStart = placement.getStartDate().plusDays(1);
+    LocalDate dayBeforePlacementStart = placement.getStartDate().minusDays(1);
 
-    Optional<Placement> optionalPlacement
-        = traineeProfile.getPlacements().stream()
-        .filter(p -> p.getTisId().equals(placementId)).findAny();
+    TraineeProfile traineeProfile = repository.findByTraineeTisId(traineeTisId);
+    return traineeProfile.getProgrammeMemberships().stream().filter(pm ->
+            pm.getStartDate().withDayOfMonth(1).isBefore(dayAfterPlacementStart)
+                && pm.getProgrammeCompletionDate().isAfter(dayBeforePlacementStart))
```

Comment:
I'm not 100% clear on this, were we seeing placements starting before placements in the data?

Reply:
Oh yes, placements often start before placements :trollface: 

Its the rule we used for placements that begin 'before' their PMs, but simply later in the same month. James has provided a slightly different rule for the PM-placement linkage work, but also accommodating this slight inconsistency in timing.